src/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.cpp

```diff
@@ -127,13 +135,74 @@ Array<State> Propagator::calculateStatesAt(const State& aState, const Array<Inst
         }
     }
 
+    // TBI: Dummy implementation at Propagator level, it should probably not live at this level
+    // Check size of this->maneuverArray_ before continuing
+
+    Array<Maneuver> forwardInstantManeuvers;
+    Array<Maneuver> backwardInstantManeuvers;
+
+    for (const Maneuver& aManeuver : this->maneuverArray_)
```

Comment:
I think this logic should be broken out, we should make a struct called PropagatedChunks or something which basically is a start time, endtime/duration, initial state, and maneuver (potentially undefined maneuver)

Reply:
Definitely, we can make the logic in there modular